tableauserverclient/server/endpoint/auth_endpoint.py

```diff
@@ -41,6 +41,9 @@ def sign_in(self, auth_req):
 
     def sign_out(self):
         url = "{0}/{1}".format(self.baseurl, 'signout')
+        # If there are no auth tokens you're already signed out. No-op
+        if not self.parent_srv._auth_token:
```

Comment:
That is what I am saying. Our logic right now is that we assume if we have a token we are signed in.  You still might fail because the token expired or something else, but from the client's standpoint, we think we are signed in. The exact same check you are doing ... just move it so we are not accessing private member variables.